src/js/autocomplete.js

```diff
@@ -135,21 +135,27 @@ function initClearButton(instance) {
 
 /**
  * @callback PopulateResults
- * @param {Array<string>} results - The results to show in the suggestions dropdown
+ * @param {Array<*>} results - The results to show in the suggestions dropdown
  * @returns {void}
  */
 
 /**
- * @callback Suggestions
+ * @callback Source
  * @param {string} query - Text which was typed into the autocomplete by the user
  * @param {PopulateResults} populateResults - Function to call when ready to update the suggestions dropdown
  * @returns {void}
 */
 
+/**
+ * @callback OptionToSuggestion
+ * @param {*|undefined} option - The option to transform into a suggestion string or undefined
+ * @returns {string} The string to display as the suggestions for this option
+*/
 
 /**
  * @typedef {Object} AutocompleteOptions
- * @property {Suggestions} source - The function which retrieves the suggestions to display
+ * @property {Source} source - The function which retrieves the suggestions to display
+ * @property {Function} [optionToSuggestion] - Function which transforms a suggestion before rendering
```

Comment:
I'm not certain that the function name `optionToSuggestion` is a good name, please let me know of any suggestions for a different name you may have

Reply:
I think `optionToSuggestedValue` is more descriptive. A "suggested value" makes sense as the return of `templates.inputValue` and is also the name of the argument for `this.suggestionTemplate`

Reply:
Maybe even `mapOptionToSuggestedValue`

Reply:
I like `mapOptionToSuggestedValue` -- I've updated the code to use that name in [`30c0c14` (#12)](https://github.com/Financial-Times/o-autocomplete/pull/12/commits/30c0c1411c73dae2c5a2dee98103ff5f299804ef)